include/openrave/kinbody.h

```diff
@@ -841,9 +841,14 @@ class OPENRAVE_API KinBody : public InterfaceBase
 
         /// \brief Sets the transform of the link regardless of kinematics
         ///
-        /// \param[in] t the new transformation
+        /// \param[in] transform the new transform of link
         virtual void SetTransform(const Transform& transform);
 
+        /// \brief Sets the transform of the link regardless of kinematics without updating parent kinbody's stamp id
+        ///
+        /// \param[in] transform the new transformation
+        virtual void SetTransformWithoutUpdateStampId(const Transform& transform);
```

Comment:
@tgn3000 this encourages bad programming if we expose it like this. there are better ways to do this....

Reply:
Admittedly if I call `SetTransform` then things still work.

I brought in `SetTransformWithoutUpdateStampId` because `Link::GetParent` copies the shared pointer and takes up noticeable amount of time in profiling. Since my FK module keeps these Links' parent body, I set their transforms and increase the body's `_nUpdateStampId` by the number of links at one time.
```
        KinBodyPtr KinBody::Link::GetParent(bool trylock=false) const {
            if( trylock ) {
                return _parent.lock();
            }
            else {
                return KinBodyPtr(_parent);
            }
        }

void KinBody::Link::SetTransform(const Transform& t)
{
    _info._t = t;
    GetParent()->_nUpdateStampId++;
}
```

Reply:
@tgn3000 you can make the  SetTransform without update time internal and pass it in to the FK solver privately.

Reply:
Now I hide it inside `ForwardKinematicsStruct::pSetSingleLinkTransformFn`.